### PR TITLE
PP-15597 Add features list in service api response

### PR DESCRIPTION
**Pull request by ralphm10**
## WHAT YOU DID

- Add features list in service api response
- Update open api spec
- Add assertions to existing integration tests to verify that empty/populated feature lists are returned
- Some minor refactoring

### CODE REVIEW

File: src/main/java/uk/gov/pay/adminusers/model/Service.java [hunk 1]
```diff
@@ -347,4 +349,13 @@ public ZonedDateTime getWentLiveDate() {
     public void setCreatedDate(ZonedDateTime createdDate) {
         this.createdDate = createdDate;
     }
+
+    public void setServiceFeatures(List<String> serviceFeatures) {
+        this.serviceFeatures = serviceFeatures;
+    }
+
+    @ArraySchema(schema = @Schema(example = "payment links"))
```
Comment (DomBelcher) [inline src/main/java/uk/gov/pay/adminusers/model/Service.java]: While there's no reason the feature names can't have spaces, I would probably expect them to be `snake_case`, but I don't really have any strong justification for it, just vibes

The best I can come up with is they can directly match names in an enum, so e.g. `PAYMENT_LINKS` could correspond to the string `payment_links` 

Even then, the examples don't really matter, they can just set the expectation of what the feature names look like

Comment (ralphm10) [reply src/main/java/uk/gov/pay/adminusers/model/Service.java]: Updated example and test values

**Review by DomBelcher — APPROVED** [review]
Very nice work 🎉
